framework/src/modules/network/network.js

```diff
@@ -95,6 +97,13 @@ module.exports = class Network {
 			{},
 			{ limit: null }
 		);
+
+		// If tried peers were stored in db retrieve pervious secret, else generate new one
+		this.secret =
+			previousPeers && previousPeers.length > 0
+				? previousPeers[0].secret
```

Comment:
We should not storing our node's secret on the peers.
Maybe it's better to not persist the secret at all.

The purpose of persisting the old secret is because generating a new secret will cause the balance of peers across buckets to change and it might cause evictions of some of our old peers; this is not a problem. Also we have tests to prove that the bucket system has approximately even distribution so there shouldn't be too many evictions (and we can definitely tolerate some; we will rediscover peers anyway).

So maybe we should not persist the secret; that way maybe we don't even need to pass it to the peer objects or put it in the peerInfo.

Reply:
I think it would be better to persist the secret rather than not, but I agree the benefit is marginal. We can put it off until 3.0.